pkg/build/api/v1beta1/defaults.go

```diff
@@ -0,0 +1,24 @@
+package v1beta1
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+)
+
+func init() {
+	err := api.Scheme.AddDefaultingFuncs(
+		func(obj *buildapi.STIBuildStrategy) {
```

Comment:
Use `STIBuildStrategy`, not `buildapi.STIBuildStrategy`- you want the defaulting to apply to the `v1beta1.STIBuildStrategy` which is the current package.